src/line_launcher/mod.rs

```diff
@@ -62,6 +68,70 @@ impl<'progression> ProgressionState<'progression> {
 const CHANNEL: Channel = Channel::Ch1;
 const VELOCITY: u8 = 100;
 
+struct NoteOffInstruction {
+    note: Note,
+    time: SystemTime,
+}
+
+struct NoteOffTriggerer {
+    receiver: Receiver<NoteOffInstruction>,
+    midi_message_sender: Arc<Mutex<MidiMessageSender>>,
```

Comment:
I think it'd be nice to wrap up the arc + mutex inside `MidiMessageSender`. It could be clonable, and the functions it exposes could internally handle dealing with the mutex.